utils/tasks.js

```diff
@@ -57,8 +58,53 @@ const buildTasks = (tasks, initialTaskArray = []) => {
   return initialTaskArray;
 };
 
+/**
+ * @param query: Record<string, string>
+ * @returns Record<filter|type|next_curosr, string> which returns
+ * - `{filter: <string>}` if query only contains key `filter`
+ * - `{type: <string>}` if query only contains key `type`
+ * - `{type: <string>, next_cursor: <string>} if query contains key `type` and `next_cursor`
+ * - `{ [filter|type|next_curosr]: <string>}` combination of any other pattern having these keys
+ */
+function convertParamsToFetchTasksQueryParams(query) {
+  const q = query.q;
+  if (isEmpty(q) || typeof q !== "string") return {};
+  return q.split(" ").reduce((acc, item) => {
+    const [key, value] = item.split(":");
+    return { ...acc, [key]: value };
+  }, {});
+}
+
+/**
+ * @param tasksModel: CollectionReference<DocumentData> it contains tasksModel collection refference
+ * @param params: Record<filter|type|cursor, string> of query parameters for building query with filtering logic, using`in` and `not-in` query operators.
+ * @returns which returns
+ * - `{whereFilterOp: <string>, status: Array<TASK_STATUS>} `if query is not paginated query
+ * - `{whereFilterOp: <string>, status: Array<TASK_STATUS>, cursor: string}` if query is paginated query
+ */
+function getFetchTasksQueryParameters(params) {
+  if (isEmpty(params)) {
+    return {};
+  }
```

Comment:
in models tasks.js isEmpty or not is already being checked, why the flag again here

Reply:
Change Done, Thanks for pointing this.

Reply:
Code Change Ref: https://github.com/Real-Dev-Squad/website-backend/pull/962/commits/e9bcb74e017662449878baec462085b076ea3199